### ISSUE TITLE
✍️ Contribution period: Shreya

### ISSUE THREAD

**Issue by alt-shreya** [ersilia-os/ersilia#642]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [x] Install the Ersilia Model Hub and test the simplest model
- [x] Write a motivation statement to work at Ersilia
- [X] Submit your first contribution to the Outreachy site

### Week 2 - Install and run an ML model

- [x] Select a model from the suggested list
- [x] Install the model in your system
- [x] Run predictions for the EML
- [x] Compare results with the Ersilia Model Hub implementation!

### Week 3 - Propose new models

- [x] Suggest a new model and document it (1)
- [x] Suggest a new model and document it (2)
- [x] Suggest a new model and document it (3)

### Week 4 - Prepare your final application

- [x] Submit the final application in the Outreachy website

**Comment by GemmaTuron** [comment]
Hi @alt-shreya 

Welcome to Ersilia! Please, make sure to complete week 1 tasks before moving on to week 2, there is still time to catch up!

**Comment by alt-shreya** [comment]
@GemmaTuron thank you!
Out of curiosity, does [this page of the docs](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/antibiotic-activity-prediction) explain tasks we need to perform in Week 2? I wanted to complete the third task for Week 1 (test the simplest model) and that's when I got a little confused.

**Comment by alt-shreya** [comment]
@GemmaTuron I ran into an error, and need your help

I'm using Fedora 37 and trying to test the simplest model (Week 1 Task 4).

In order to fetch the model, I executed this command:
```bash 
ersilia fetch retrosynthetic-accessibility 
```

 and got this error:

```bash
/bin/sh: line 1: shasum: command not found
Command '<<address of the file>> shasum -a 256 data.h5;' returned non-zero exit status 127. 
```

I also checked the log files in order to understand where the error may have been, but the log file is empty.

Any help will be appreciated. Thank you!

**Comment by GemmaTuron** [comment]
Hi @alt-shreya 

Do you have git-lfs installed? it seems you were not able to download the .h5 file initially

**Comment by alt-shreya** [comment]
# My Motivation to Work at Ersilia

Dear Gemma Turon,

In order to explain my motivation, I’d like to start this letter with an anecdote. When I was in kindergarten, my school conducted a free health checkup for the students. This was the first time I had been to a clinic. And I was terrified. To me, the unknown doctors behind their masks seemed like the minions of an unknown mastermind. Eighteen years later, I am glad to report the fear I once held has given its way to a sense of deep respect for the healthcare industry. That is my primary motivation behind applying for this internship.

As an aspiring AI professional, I am eager to gain direct exposure to the ML industry while contributing to a cause that I am passionate about – making medical research more accessible. I resonate with your accessible and open approach to research, as well as your commitment to building sustainable collaborations with leading local entities. I believe that collaboration is essential to achieving breakthroughs in science and technology, and I am highly enthusiastic to join an open-source software community that shares this value.

Throughout my academic and personal journey, I have developed a strong interest in machine learning and artificial intelligence. During my Bachelor’s degree, I was determined to make the most of this interest to make an impact in the healthcare industry. I led a team of four to build a prototype for a contactless authentication device for a post-COVID world. This experience taught me important lessons on making technology accessible, combining technical skills with a passion for solving real-world problems, perseverance, and leadership.

Growing up in India, I have always understood the importance of making science affordable and accessible to everyone. As an intern at the Ersilia Open Source Initiative, I believe I would have the opportunity to contribute to this cause while honing my skills in machine learning under the guidance of experienced mentors. I am detail-oriented and have honed my research skills through hours of academic work. Additionally, I also volunteered extensively during the COVID-19 pandemic to help raise funds for those in need. Having graduated into the workforce under extraordinary circumstances, I quickly learnt to thrive in remote work environments and enjoy collaborating with a diverse group of people.

In conclusion, I am honoured to have the opportunity to apply for the internship position at Ersilia Open Source Initiative, and I believe that my experience, skills, and passion align well with the values and goals of your non-profit organisation. Thank you for considering my application. I would be happy to discuss my candidature in detail via the following platforms:

**Email:** shreyakumar31@gmail.com
**LinkedIn:** https://www.linkedin.com/in/alt-shreya
**GitHub:** https://www.github.com/alt-shreya

**Comment by alt-shreya** [comment]
# Week 2 Log

## Main Tasks
- [x]  Select a model from the suggested list
- [x]  Install the model in your system
- [ ]  Run predictions for the EML
- [ ]  Compare results with the Ersilia Model Hub implementation!

## Things that Motivate me 
These tasks provide a hands-on glimpse of my responsibilities during the internship; finding, testing, debugging and implementing open source code within Ersilia.

## Task Report
### Task 1: Selecting a Model


- At first glance, STOUT seems really interesting. When I was in high school trying to understand IUPAC naming conventions, I wondered what it would be like to have an automatic name generator which could make life easy for me.
- For context, IUPAC names are a standard set of names for compounds decided by the Internatinoal Union of Pure and Applied Chemistry. Under the IUPAC convention, the nomenclature of vanillin, the compound that lends vanilla essence its flavour, would be written as ```4-hydroxy-3-methoxybenzaldehyde```.

- SMILES strings are a line notation that represent the structure of the compound. The SMILES for the compound vanillin would be ```c1(C=O)cc(OC)c(O)cc1```

- Aside from personal life, I think this could be significant in minimising errors in drug discovery.
- I am curious and highly motivated to implement this.
- Taking a deeper dive into the research paper, I am intrigued by their approach towards representing chemical entities and string conversion.
- I'd like to see this in action.

### Installing 

- Step1: created a separate conda environment by running the following steps:

```bash
conda create --name STOUT python=3.8 
conda activate STOUT
conda install -c decimer stout-pypi
```
the last step resulted in an error as there were some incompatible packages. So I decided to try another approach:  installing repo from Git, like so:

```pip install git+https://github.com/Kohulan/Smiles-TO-iUpac-Translator.git```

This successfully built STOUT. 

However, when I tried to import Python, it gave me a module not found error. I realised I had to install the dependencies independently, something I had inadvertently missed out on. 
I install the requirements using

```pip install -r Python_requirements.txt```

This approach also led to some errors. On reading the error descriptions, I understood it is due to the proper version of Tensorflow not being installed. To fix this, I installed Tensorflow out of my conda environment. 

As of writing this, TensorFlow is still installing on my system. I will update this space here with further inputs. 

### Next Steps
My next steps would be to 
- run the sample code provided on the STOUT documentation 
- run predictions for the EML 
- ask fellow applicants who selected this model about errors they faced

**Comment by GemmaTuron** [comment]
Hi @alt-shreya 

Good start, let's complete these tasks before moving to week 3!

**Comment by masroor07** [comment]
> 
Hey @alt-shreya,
Are you providing the right input to the function when you calling it?

**Comment by alt-shreya** [comment]
@masroor07 yes

**Comment by masroor07** [comment]
> @GemmaTuron I used this command to write my input:
> 
> ```python
> SMILES = "CN1C=NC2=C1C(=O)N(C(=O)N2C)C"
> 

> IUPAC_name = translate_forward(SMILES)```
> ```

Make sure it is a valid smile

> @GemmaTuron I used this command to write my input:
> 
> ```python
> SMILES = "CN1C=NC2=C1C(=O)N(C(=O)N2C)C"
> IUPAC_name = translate_forward(SMILES)```
> ```

Could you try running it for a different SMILE from the eml file and check whether you get the same error?

**Comment by alt-shreya** [comment]
Thank you @masroor07 and @GemmaTuron, I tried the same formula with some other SMILES formulae, and tehy worked. I will be updating this space with my progress now.

**Comment by alt-shreya** [comment]
### Update 3: Predicting EML file

After a little help from my friends who also happened to be working on this model, I finally made my first translation. Once again, it was the compound vanillin. Following is the code I entered:

```python
from STOUT import translate_forward, translate_reverse
import pandas as pd
import numpy as np

SMILES = "SMILES = 'c1(C=O)cc(OC)c(O)cc1'
IUPAC_name = translate_forward(SMILES)
print("IUPAC name of "+SMILES+" is: "+IUPAC_name)

# IUPAC name to SMILES translation

IUPAC = "4-hydroxy-3-methoxybenzaldehyde"
SMILES = translate_reverse(IUPAC)
print("SMILES of "+IUPAC_name+" is: "+SMILES)
```
Here is the file with the outputs
[Output.txt](https://github.com/ersilia-os/ersilia/files/11043121/Output.txt)

Then, I had only to perform the same tasks on all 422 rows of the ```eml_canonical.csv``` file. In order to do so, here are the steps I undertook:

#### Running Predictions for the EML
```python
# converting dataframe to an a list of strings

df = pd.read_csv('/filepath/eml_canonical.csv')
eml_smiles = df["smiles"]
eml_smiles_list= eml_smiles.tolist()
IUPAC_list = []

# for loop to convert every entry in the CSV file to an IUPAC name
for smile in eml_smiles_list:
  IUPAC_name = translate_forward(smile)
  IUPAC_list.append(IUPAC_name)

# converting IUPAC string to a CSV
new_dict = {"smiles": eml_smiles_list, "IUPAC": IUPAC_list}
output = pd.DataFrame(new_dict)
output.to_csv('STOUT_OUT.csv')
```
As of now, the code is still running, perhaps due to the sheer number of entries in the file. More updates to follow:

### Next Steps
If this doesn't get resolved in a few hours, I will create a separate file with fewer entries and run that through my ```for``` loop. This ought to give me a sample of outputs, enough to build an understanding of the STOUT model.

**Comment by masroor07** [comment]
> Thank you @masroor07 and @GemmaTuron, I tried the same formula with some other SMILES formulae, and they worked. I will be updating this space with my progress now.
Hi @GemmaTuron ,
I tried looking for the SMILE "CN1C=NC2=C1C(=O)N(C(=O)N2C)C" in the EML, could not find it there.

**Comment by masroor07** [comment]
> > Thank you @masroor07 and @GemmaTuron, I tried the same formula with some other SMILES formulae, and they worked. I will be updating this space with my progress now.

 Hi @GemmaTuron ,
 I tried looking for the SMILE "CN1C=NC2=C1C(=O)N(C(=O)N2C)C" in the EML, could not find it there.

**Comment by alt-shreya** [comment]
@masroor07 I think it's some form of caffeine, but yeah, it doesn't exist in the EML file.

**Comment by alt-shreya** [comment]
### Update 5: Comparing the files
Presenting my findings in this table

| **SMILES** | **Ersilia Model** | **STOUT Model** |
|---|---|---|
| **CC(=O)N[C@@H](CS)C(O)=O** | **(2R)-2-acetamido-3-sulfanylpropanoicacid;;;** | **(2R)-2-acetamido-3-sulfanylpropanoicacid;;;** |
| **CC(=O)Oc1ccccc1C(O)=O** | **2-acetyloxybenzoicacid;;;** | **2-acetyloxybenzoicacid;;;** |
| **NC1=NC(=O)c2ncn(COCCO)c2N1** | **2-amino-9-(2-hydroxyethoxymethyl)-3H-purin-6-one;;;** | **2-amino-9-(2-hydroxyethoxymethyl)-3H-purin-6-one;;;** |
| **NC(N)=NC(=O)c1nc(Cl)c(N)nc1N** | ;3,5-diamino-2-chloro-N-(diaminomethylidene)-2H-pyrazine-6-carboxamide;;; | 3,5-diamino-6-chloro-N-(diaminomethylidene)pyrazine-2-carboxamide;;; |
| **CCCCc1oc2ccccc2c1C(=O)c3cc(I)c(OCCN(CC)CC)c(I)c3** | 2-butyl-3-[4-[2-(diethylamino)ethoxy]-3,5-diiodocyclohexa-1,4-dien-1-yl]chromen-4-one;;; | (2-butyl-1-benzofuran-3-yl)-[4-[2-(diethylamino)ethoxy]-3,5-diiodophenyl]methanone;; |

**Comment by alt-shreya** [comment]
# Proposed Model 1: DTI2VEC

### Model Name: 
DTi2Vec: Drug-target interaction prediction using network embedding and ensemble learning

### Model Summary
DTi2Vec identifies DTIs using network representation learning and ensemble learning techniques. It constructs the heterogeneous network, and then it automatically generates features for each drug and target using the nodes embedding technique. DTi2Vec is a simple yet effective method that provides high DTI prediction performance while being scalable and efficient in computation, translating into a powerful drug repositioning tool.


### Datasets
* Nuclear Receptor dataset (nr),
* G-protein-coupled receptor (gpcr),
* Ion Channel (ic),
* Enzyme (e)
* FDA_DrugBank (DrugBank) which each one of them has all required data of drug-target interactions (in Adjacency matrix and edgelist format) and drug-drug similarity and target-target similarity in (square matrix format)

### Slug: dti2vec

### Tag: machine-learning, cheminformatics, drug-repurposing, node2vec-embeddings, boosting-ensemble

### Task: Classification

### Package Dependencies:
* gensim (for node2vec code)
* numpy
* Scikit-learn
* imblearn
* pandas
* xgboost

#### Publication https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8459562/

#### Source Code [Github Respository](https://github.com/MahaThafar/DTi2Vec)

#### License: Creative Commons

**Comment by alt-shreya** [comment]
# Proposed Model 2: DeepCC

#### Model Name: DeepCC
#### Model Summary: a novel deep learning-based framework for cancer molecular subtype classification
#### Model Description:
A novel supervised cancer classification framework, deep cancer subtype classification (DeepCC), based on deep learning of functional spectra quantifying activities of biological pathways. DeepCC provides a novel cancer classification framework that is platform independent, robust to missing data, and can be used for single sample prediction facilitating clinical implementation of cancer molecular subtyping.


#### Datasets
* TANSBIG, UNT, UPP, and NKI
* TCGA CRC set - level 3 RNASeq data
* MXNet

#### Slug: deepcc

#### Tag: molecular subtyping, individualized therapy
#### Task: Classification

#### Language: R, Python

#### Package Dependencies
* keras
* R (version 3.3, or higher)
* C
* MXNet (version 0.10, or higher)

#### [Publication ](https://www.nature.com/articles/s41389-019-0157-8.pdf?error=cookies_not_supported&code=d51aa87c-af88-4144-ba50-96982967ec13)
#### Source Code 
[Github Respository](https://github.com/CityUHK-CompBio/DeepCC)

#### Reference
Gao, F., Wang, W., Tan, M. et al. DeepCC: a novel deep learning-based framework for cancer molecular subtype classification. Oncogenesis 8, 44 (2019). https://doi.org/10.1038/s41389-019-0157-8

#### License
MIT License

**Comment by GemmaTuron** [comment]
Hi @alt-shreya 

Thanks for these suggestions, these are good models but mostly focused on cancer, which is out of the scope for us at this moment.
Let's wrap up by preparing your final application
